src/java.base/share/classes/java/lang/invoke/InnerClassLambdaMetafactory.java

```diff
@@ -442,14 +442,24 @@ private void generateQuotedFieldInitializer(CodeBuilder cob) {
         NameAndTypeEntry natMH = cp.nameAndTypeEntry(DEFAULT_NAME, CD_MethodHandle);
         // push the receiver on the stack for operand of put field instruction
         cob.aload(0)
+        // load class data: CodeReflectionSupport.HANDLE_MAKE_QUOTED and quotableOpGetter
            .ldc(cp.constantDynamicEntry(cp.bsmEntry(bsmDataAt, List.of(cp.intEntry(2))), natMH))
-        // load op string from field
            .ldc(cp.constantDynamicEntry(cp.bsmEntry(bsmDataAt, List.of(cp.intEntry(1))), natMH));
         MethodType mtype = quotableOpGetterInfo.getMethodType();
         if (quotableOpGetterInfo.getReferenceKind() != MethodHandleInfo.REF_invokeStatic) {
             mtype = mtype.insertParameterTypes(0, implClass);
         }
+        if (quotableOpGetterInfo.getMethodType().parameterList().equals(List.of(CodeReflectionSupport.OP_FACTORY_CLASS,
+                CodeReflectionSupport.TYPE_ELEMENT_FACTORY_CLASS))) {
```

Comment:
If these fields are public/static, why don't we load them from inside the compiler-generated method, instead of injecting them here?

Reply:
I am assuming by the compiler-generated method you mean `opMethod`. The `opMethod` has two params (OpFactory, TypeElementFactory), so loading these fields is necessary before invoking it.

Reply:
We can turn opMethod into one with no param and load these fields inside its body. But this removes the flexibility we have now (i.e. we can choose the OpFactory and TypeElementFactory we want to pass).

Reply:
> We can turn opMethod into one with no param and load these fields inside its body.

That is what I was asking about -- while I agree what we have seems more flexible, we don't seem to use this flexibility anywhere -- as the method that turns a `Quotable` into a `Quoted` works with hardwired op and type factories. But maybe this has been already discussed -- in which case it's fine.

Reply:
Alternatively use a consistent the method signature with the parameters even if they are not used. Uniformity in the method signature is i think a nice property we should aim for, but could be followed up later.

Reply:
Yep - I agree we can follow up later.